apm-commons/apm-datacarrier/src/main/java/org/apache/skywalking/apm/commons/datacarrier/consumer/MultipleChannelsConsumer.java

```diff
@@ -30,12 +31,13 @@
 public class MultipleChannelsConsumer extends Thread {
     private volatile boolean running;
     private volatile ArrayList<Group> consumeTargets;
-    private volatile long size;
+    private final AtomicLong size;
```

Comment:
This is just for changed value accessable from other consumer thread. No requirement for AtomicLong.